### PR TITLE
moving the SQL stuff away from the Member constructor and fixing some broken unit tests

### PR DESCRIPTION
**Pull request by whirm**
Do not merge it yet for the love of the FSM!

### CODE REVIEW

File: dispersy.py [hunk 1]
```diff
@@ -551,7 +551,45 @@ def get_member(self, *argv, **kwargs):
...
+
+            # Store the key only if we have the private key pair and it didn't come from the DB
+            if private_key and not private_key_from_db:
+                self.database.execute(u"INSERT INTO private_key (member, private_key) VALUES (?, ?)",
+                                      (database_id, buffer(private_key)))
+            elif private_key_from_db:
+                private_key = private_key_from_db
+                key = self.crypto.key_from_private_bin(private_key)
+
+            member = Member(self, key, database_id)
```
Comment (NielsZeilemaker) [inline dispersy.py:592]: The sql is now moved, but it still is completely unreadable :smile: However, that can be fixed if we start to add the private key column to the member table.

**Comment by NielsZeilemaker** [comment]
Could you move the sql from the constructor of the DummyMember as well?

File: dispersy.py (hunk 1 above)
Comment (whirm) [inline dispersy.py:592]: yeah, let's fix the unit tests first :)

**Comment by whirm** [comment]
retest this please

**Comment by whirm** [comment]
@NielsZeilemaker Ready to merge

File: dispersy.py [hunk 2]
```diff
@@ -1589,7 +1635,10 @@ def on_incoming_packets(self, packets, cache=True, timestamp=0.0):
 
         self._statistics.received_count += len(packets)
 
-        sort_key = lambda tup: (tup[1][2:22], tup[1][1], tup[1][22])  # community ID, community version, message meta type
+        # Ugly hack to sort the identity messages before any other to avoid sending missing identity requests
+        # for identities we have already received but not processed yet. (248 == identity message ID)
+        #                                           /--------------------------\
+        sort_key = lambda tup: (tup[1][2:22], tup[1][1], 0 if tup[1][22] ==  248 else tup[1][22])  # community ID, community version, message meta type
```
Comment (NielsZeilemaker) [inline dispersy.py]: Did you test this? Shouldn't we compare it to chr(248) ?

**Pull request merged**